drivers/mtd/nand/raw/pl35x-nand-controller.c

```diff
@@ -795,6 +795,22 @@ static int pl35x_nfc_exec_op(struct nand_chip *chip,
 				      op, check_only);
 }
 
+static unsigned long get_special_timing_freq(const struct device_node *np, const int timing_mode)
+{
+	int len, i;
+
+	const __be32 *memclk_timing_freq = of_get_property(np, "memclk-timing-frequency", &len);
+
+	if (memclk_timing_freq) {
+		for (i = 0; i < len / sizeof(u32) / 2; i++) {
+			if (timing_mode == be32_to_cpu(memclk_timing_freq[i * 2])) {
+				return be32_to_cpu(memclk_timing_freq[i * 2 + 1]);
```

Comment:
In theory this could index beyond the end of the array depending on the value of 'len' and how the 'len/sizeof(u32)/2' math in the for loop gets truncated.

Not sure yet if there is a better way to iterate through this array. I'll have to do some digging through the kernel to see if there are other examples.